### PR TITLE
002 view role detail

### PR DESCRIPTION
**Pull request by Brigid481**
updated job role table to contain responsibilities, description and link to spec. We added new get job Roles by ID endpoint. We did unit and Integration Tests.

### CODE REVIEW

**Review by JakubRS7 — APPROVED** [review]

File: src/main/java/org/example/Exceptions/Entity.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.example.Exceptions;
+
+public enum Entity {
```
Comment (boydstepkainos) [inline src/main/java/org/example/Exceptions/Entity.java]: Entity? The name is throwing me off

File: src/main/java/org/example/TestApplication.java [hunk 2]
```diff
@@ -44,6 +44,9 @@ public void run(final TestConfiguration configuration,
         Key jwtKey = Jwts.SIG.HS256.key().build();
         environment.jersey().register(new AuthController(new AuthService(
                 new AuthDao(), jwtKey)));
+
     }
 
+
```
Comment (boydstepkainos) [inline src/main/java/org/example/TestApplication.java]: Why?

Comment (RaggedyRach) [reply src/main/java/org/example/TestApplication.java]: spaces removed

File: src/main/java/org/example/Exceptions/Entity.java (hunk 1 above)
Comment (RaggedyRach) [reply src/main/java/org/example/Exceptions/Entity.java]: re named to Error entity as it refers to the entity the error occurred with

File: src/main/java/org/example/models/JobRoleDetailResponse.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package org.example.models;
+
+import java.sql.Date;
+
+public class JobRoleDetailResponse extends JobRoleResponse {
+
+    /*
+    instantiates a job role spec Url in the Job Role Response
```
Comment (boydstepkainos) [inline src/main/java/org/example/models/JobRoleDetailResponse.java]: Not just a url

File: src/main/java/org/example/services/JobRoleService.java [hunk 4]
```diff
@@ -14,8 +17,21 @@ public JobRoleService(final JobRoleDao jobRoleDao) {
...
         return jobRoleDao.getOpenJobRoles();
     }
+
+    public JobRoleDetailResponse getJobRoleById(final int id)
+            throws SQLException, DatabaseConnectionException,
+            DoesNotExistException {
+        JobRoleDetailResponse jobRoleDetailResponse
+                = jobRoleDao.getJobRoleByID(id);
+        if (jobRoleDetailResponse == null) {
+            throw new DoesNotExistException(Entity.JOBROLEDETAILRESPONSE);
```
Comment (boydstepkainos) [inline src/main/java/org/example/services/JobRoleService.java]: This enum is hard to read

**Review by JakubRS7 — APPROVED** [review]

**Review by boydstepkainos — APPROVED** [review]

File: src/main/java/org/example/services/JobRoleService.java (hunk 4 above)
Comment (RaggedyRach) [reply src/main/java/org/example/services/JobRoleService.java]: change to screaming snake case

**Pull request merged**